server/ethereum/ethereum.test.js

```diff
@@ -29,18 +30,87 @@ afterAll(() => {
   web3.currentProvider.connection.close()
 })
 
-// NOTE - this isn't really testing anything yet, just showing how to fire an event and getting the testing framework working
-test('firing an event pins a file', async done => {
+test('watcher pins file from registerData function', async done => {
   const testKey = web3.utils.fromAscii('testKey')
-  const testVal = { dag: 'killinnnnnn it' }
-  const hash = await node.dag.put(testVal)
+  const hash = await node.dag.put(dag)
 
   registerWatcher(contract)
   contract.methods
     .registerData(testKey, hash.toBaseEncodedString())
     .send({ from: accounts[0] }, () => {
-      setTimeout(() => {
+      setTimeout(async () => {
+        const pins = await node.pin.ls()
+        const match = pins.find(item => {
+          return item.hash === hash.toBaseEncodedString()
+        })
+        expect(match).toBeDefined()
         done()
       }, 2000)
     })
 })
+
+test('handlePinHashEvent pins file of cid it was passed', async done => {
+  dag.testKey = 'handlePinHashEventTestVal'
+  const hash = await node.dag.put(dag)
+  const eventObj = {
+    returnValues: {
+      cid: hash.toBaseEncodedString()
+    }
+  }
+
+  const res = await handlePinHashEvent(null, eventObj)
+  expect(res[0].hash).toBe(hash.toBaseEncodedString())
+  done()
+})
+
+test('handlePinHashEvent throws error with empty params', async done => {
+  try {
+    await handlePinHashEvent()
+    expect(true).toBe(false)
+    done()
+  } catch (error) {
+    expect(error).toBeDefined()
+    done()
+  }
+})
+
+test('getContract returns a contract', async done => {
+  const contract = getContract(
+    demoSmartContractJson1,
+    demoSmartContractJson1.address
+  )
+  expect(contract._address).toBe(demoSmartContractJson1.address)
+  done()
+})
+
+test('getContract throws when an invalid contract is passed', async done => {
+  demoSmartContractJson1.address = '0x7505462c30102eBCDA555446c3807362AeFEfc8r'
+  const badCall = () => {
+    return getContract(
+      demoSmartContractJson1,
+      '0x7505462c30102eBCDA555446c3807362AeFEfc8r'
+    )
+  }
+
+  expect(badCall).toThrow()
+  done()
+})
+
+// Uncomment this when timeout in ipfs.dag.get is working. Will need to adjust jest.setTimeout.Timeout past the dag.get timeout time.
+
+// test('handlePinHashEvent throws and error with an invalid cid', async done => {
```

Comment:
might help to just rename this test: "handlePinHashEvent throws an error after X seconds after the cid is unavailable on the network".

We could write another test to throwing an error immediately in the `cid` is actually malfored (e.g. `xyz123`).